auditor/src/client/mod.rs

```diff
@@ -174,6 +175,139 @@ impl Default for AuditorClientBuilder {
     }
 }
 
+#[derive(serde::Deserialize, Debug, Default, Clone)]
+pub struct DateTimeUtcWrapper(pub DateTime<Utc>);
+
+impl Serialize for DateTimeUtcWrapper {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        serializer.serialize_str(&self.0.to_rfc3339())
+    }
+}
+
+/// The 'QueryParameters' is used to build query parameters which allows to query records from
+/// the database using advanced_query function.
+#[derive(serde::Deserialize, serde::Serialize, Debug, Default, Clone)]
+pub struct QueryParameters {
+    pub start_time: Option<TimeOperator>,
+    pub stop_time: Option<TimeOperator>,
+    pub runtime: Option<TimeOperator>,
+}
+
+impl Default for QueryBuilder {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+#[derive(serde::Deserialize, Debug, Clone)]
+pub enum TimeValue {
+    Datetime(DateTimeUtcWrapper),
+    Runtime(u64),
+}
+
+impl Serialize for TimeValue {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        match self {
+            TimeValue::Datetime(datetime) => datetime.serialize(serializer),
+            TimeValue::Runtime(runtime) => runtime.serialize(serializer),
+        }
+    }
+}
+
+/// The 'TimeOperator' struct is used to specify the operators on the query parameters like (gt ->
+/// greater than, lt -> lesser than, gte -> greater than equals, lte -> lesser than equals)
```

Comment:
Documentation: Use either backticks (for proper rendering with a monospace font) or remove the normal hyphens. Also, maybe you can improve the formatting of the gt/lt/... explanation - or leave it out here and document the individual fields below.